### ISSUE TITLE
Google login does not generate a refresh token on second and subsequent login; throws exception

### ISSUE THREAD

**Issue by lukehutch** [serverpod/serverpod#754]
**Describe the bug**

If a user logs in using Google more than once, the second and subsequent times they are not provided with a refresh token, which causes the following exception:

```
METHOD CALL: serverpod_auth.google.authenticateWithServerAuthCode duration: 219ms numQueries: 0 authenticatedUser: null
'package:googleapis_auth/src/auth_http_utils.dart': Failed assertion: line 98 pos 16: 'credentials.refreshToken != null': is not true.
auth_http_utils.dart:98

#0      _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:51:61)
#1      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:40:5)
#2      new AutoRefreshingClient
auth_http_utils.dart:98
#3      _GoogleUtils.clientViaClientSecretAndCode
google_endpoint.dart:237
<asynchronous suspension>
#4      GoogleEndpoint.authenticateWithServerAuthCode
google_endpoint.dart:33
<asynchronous suspension>
#5      EndpointDispatch.handleUriCall
endpoint_dispatch.dart:111
<asynchronous suspension>
#6      Server._handleRequest
server.dart:243
<asynchronous suspension>
```

The issue is explained here:

https://stackoverflow.com/a/10857806/3950982

The fix is to add `prompt=consent&access_type=offline` to the request, to require the user to consent to Google login each time.

These parameters are omitted in `obtainAccessCredentialsViaCodeExchange` in `googleapis_auth-1.3.1/lib/src/oauth2_flows/auth_code.dart`.

I will submit a PR that adds these parameters.

**Comment by lukehutch** [comment]
The corresponding bug in google auth libraries: https://github.com/google/googleapis.dart/issues/487

**Comment by lukehutch** [comment]
The correct fix is #757.
